### DISCUSSION TITLE
Adding microtiming/unquantized recording

### DISCUSSION THREAD

**Discussion by bogdanoctav** [bangcorrupt/hacktribe#304] [Ideas]
Hey! Found out about your hacktribe project today. And I'm quite excited about it. Finally someone takes it on!

Here is my biggest wish for the electribe 2, by far:
The ability to record off the grid/unquantized/microtiming. I'm working with the electribe for some 8 years, and this is the one thing that made me use it less than it could.
As a percussionist, I often find myself thinking of musical phrases that slightly shift off the beat, or combine a binary and a ternary grid.

Is there a possibility for you to do that?
Would make me want to sponsor you directly. :D

**Comment by bangcorrupt** [comment]
I think this might be possible at some point.  A lot of the functionality is already there; gate time, groove templates and modulations are all updated at 96 ticks per step.  It may be a while before I work on it, but it's a popular request and a feature I want.  This is also discussed in #36.

**Reply by bogdanoctav** [reply]
Wonderful! I will try to send some money for that.
Especially being able to live record off the grid (either selecting different quantizations, or enabling something like the substeps on the Novation Circuit, or direct unquantized recording as on the MPC) would be a total gamechanger.
There is also the Elektron approach to this, but I don't know too much about that.
If I can help with that, count me in.

**Reply by bangcorrupt** [reply]
The closest thing at the moment is custom groove templates. If you're comfortable programming in Python, you can try editing some templates by modifying [this script](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2_groove_template.py). This will let you move some steps off beat, but there is currently no way of saving it.  There is also the Lag IFX to delay an entire part, but it will use up a voice of polyphony.

See #34 and #113.

Thank you very much for sponsoring Hacktribe!

**Comment by bogdanoctav** [comment]
Hey! Thanks. Programming groove templates wouldn't do it for me, as I really want to be able to record stuff live, off the grid. J Dilla style, kind of. It doesn't help to have to select from a list of groove templates for stuff that I'm coming up with on the spot.More creative freedom in being able to record the phrases that I play live into the electribe. I'm hopeful that you can pull that off. :D Big up!
